### PR TITLE
Remove xtend.lib dependency

### PR DESCRIPTION
**Pull request by laeubi**
Currently there are some small leftovers to xtext/xtend that seem no longer needed. This currently ties lemminx unnecessarily to xtext 

FYI @szarnekow

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/CodeLensKindCapabilities.java [hunk 1]
```diff
@@ -13,8 +13,7 @@
 
 import java.util.List;
 
-import org.eclipse.xtext.xbase.lib.Pure;
-import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+import org.eclipse.lsp4j.jsonrpc.util.ToStringBuilder;
```
Comment (szarnekow) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/CodeLensKindCapabilities.java:16]: This was not available in lsp4j 0.20.x

Comment (laeubi) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/CodeLensKindCapabilities.java:16]: Hm... do one really needs this helper class here? It really seems not suing all the facy stuff here any maybe a standard java StingBuilder would suffice?

Comment (szarnekow) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/client/CodeLensKindCapabilities.java:16]: It can be inlined for sure.

**Comment by laeubi** [comment]
@szarnekow as you are more familiar with xtext/xtend could you probably help out to resolve this?

**Comment by angelozerr** [comment]
If I remember we have this dependecy only for ToStringBuilder.

I would like to avoid bumping lsp4ij because we need to take care of binary lemminx support and our team have no time for the moment to do that.

I suggest that you create you copy paste the xtend ToStribgBuilder in the lemminx utils package.

**Comment by laeubi** [comment]
> I suggest that you copy paste the xtend ToStribgBuilder in the lemminx utils package.

Yes that's what I think @szarnekow recommended, but maybe thats a better task for someone familiar with xtext / lemminx.

**Comment by angelozerr** [comment]
> > I suggest that you copy paste the xtend ToStribgBuilder in the lemminx utils package.
> 
> Yes that's what I think @szarnekow recommended, but maybe thats a better task for someone familiar with xtext / lemminx.

We will not have time to do that.

**Comment by laeubi** [comment]
> > > I suggest that you copy paste the xtend ToStribgBuilder in the lemminx utils package.
> > 
> > 
> > Yes that's what I think @szarnekow recommended, but maybe thats a better task for someone familiar with xtext / lemminx.
> 
> We will not have time to do that.

Can you give some insights about "we"? How can one help with bring lemminx forward in terms of maintenance, maybe @tfroment can help out and make this project get more resources (of whatever kind required), because XML is an important part in the IDE, especially for m2e is also dependent on this.

**Comment by laeubi** [comment]
@angelozerr I now copied this from lsp4j and adjusted for java 1.8 compatibility (some vars to actual type and a missing generic).

**Comment by laeubi** [comment]
I verified this by 

```
				<exclusion>
					<groupId>org.eclipse.xtend</groupId>
					<artifactId>org.eclipse.xtend.lib</artifactId>
				</exclusion>
```

then the build succeeds as well, I think we just need to keep the dependency so it show up in the uber-jar

**Comment by szarnekow** [comment]
Certainly an interesting case, since we worked hard to remove the Xbase dependency from lsp4j in the past. According to maven central, it's not part used by lsp4 0.21.1 and newer. Maybe upgrade the lsp version, instead?

**Comment by laeubi** [comment]
> Maybe upgrade the lsp version, instead?

See
- https://github.com/eclipse-lemminx/lemminx/pull/1629

it seems Java 1.8 compatibility is still a concern.

**Comment by angelozerr** [comment]
> > > > I suggest that you copy paste the xtend ToStribgBuilder in the lemminx utils package.
> > > 
> > > 
> > > Yes that's what I think @szarnekow recommended, but maybe thats a better task for someone familiar with xtext / lemminx.
> > 
> > We will not have time to do that.
> 
> Can you give some insights about "we"? 

@fbricon @rgrunber @datho7561 and me, but badly we are very busy with other projects since lemminx is enough stable and provide features expected for vscode xml

> How can one help with bring lemminx forward in terms of maintenance, maybe @tfroment can help out and make this project get more resources (of whatever kind required), because XML is an important part in the IDE, especially for m2e is also dependent on this.

Any contribution are welcome and we will try our best to review prs

**Comment by tfroment** [comment]
Hi @angelozerr, thank you @laeubi 
The Eclipse IDE working group depends on the planning council to prioritize funded development efforts (outlined at https://eclipseide.org/working-group/development-funding-initiative/) and listed on this board https://gitlab.eclipse.org/eclipse-wg/ide-wg/ide-wg-dev-funded-efforts/ide-wg-dev-funded-program-planning-council-top-issues/-/boards/1208. If the planning council deems it a high priority, it can then be considered through this process. I will share it with them.

**Comment by laeubi** [comment]
Build failures seem to be the usual instabilities, so this is ready to be merged from my side.

**Comment by angelozerr** [comment]
LGTM. Thanks @laeubi !

**Comment by rgrunber** [comment]
FWIW the same was done on https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2416 .

**Pull request merged**